src/cmd/turnkey/pkg/wallets.go

```diff
@@ -0,0 +1,350 @@
+						path = *standardPath
+						break
+					}
+				}
+			}
+		}
+
+		activity := string(models.ActivityTypeCreateWalletAccounts)
+
+		params := wallets.NewCreateWalletAccountsParams()
+		params.SetBody(&models.CreateWalletAccountsRequest{
+			OrganizationID: &Organization,
+			Parameters: &models.CreateWalletAccountsIntent{
+				WalletID: wallet.WalletID,
+				Accounts: []*models.WalletAccountParams{
+					{
+						AddressFormat: &addressFormat,
+						Curve:         &curve,
+						PathFormat:    &pathFormat,
+						Path:          &path,
+					},
+				},
+			},
+			TimestampMs: util.RequestTimestamp(),
+			Type:        &activity,
+		})
+
+		if err := params.Body.Validate(nil); err != nil {
+			OutputError(eris.Wrap(err, "request validation failed"))
+		}
+
+		resp, err := APIClient.V0().Wallets.CreateWalletAccounts(params, APIClient.Authenticator)
+		if err != nil {
+			OutputError(eris.Wrap(err, "request failed"))
+		}
+
+		if !resp.IsSuccess() {
+			OutputError(eris.Errorf("failed to create wallet account: %s", resp.Error()))
+		}
+
+		Output(resp.Payload)
+	},
+}
+
+var walletAccountsListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Return accounts for the wallet",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if walletName == "" {
+			OutputError(eris.New("name for wallet must be specified"))
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		wallet, err := lookupWalletByName(walletName)
+		if err != nil {
+			OutputError(eris.Wrap(err, "failed to lookup wallet"))
+		}
+
+		params := wallets.NewGetWalletAccountsParams()
+
+		params.SetBody(&models.GetWalletAccountsRequest{
+			OrganizationID: &Organization,
+			WalletID:       wallet.WalletID,
+		})
+
+		if err := params.Body.Validate(nil); err != nil {
+			OutputError(eris.Wrap(err, "request validation failed"))
+		}
+
+		resp, err := APIClient.V0().Wallets.GetWalletAccounts(params, APIClient.Authenticator)
+		if err != nil {
+			OutputError(eris.Wrap(err, "request failed"))
+		}
+
+		if !resp.IsSuccess() {
+			OutputError(eris.Errorf("failed to list wallets: %s", resp.Error()))
+		}
+
+		Output(resp.Payload)
+	},
+}
+
+func lookupWalletByName(name string) (*models.Wallet, error) {
+	params := wallets.NewGetWalletsParams()
+
+	params.SetBody(&models.GetWalletsRequest{
+		OrganizationID: &Organization,
+	})
+
+	if err := params.Body.Validate(nil); err != nil {
+		OutputError(eris.Wrap(err, "request validation failed"))
+	}
+
+	resp, err := APIClient.V0().Wallets.GetWallets(params, APIClient.Authenticator)
+	if err != nil {
+		OutputError(eris.Wrap(err, "request failed"))
+	}
+
+	if !resp.IsSuccess() {
+		OutputError(eris.Errorf("failed to list wallets: %s", resp.Error()))
+	}
+
+	for _, wallet := range resp.Payload.Wallets {
+		if *wallet.WalletName == name {
+			return wallet, nil
+		}
+	}
+
+	return nil, eris.Errorf("wallet name %q not found in list of wallets", name)
+}
+
+func listAccountsForWallet(walletID *string) ([]*models.WalletAccount, error) {
```

Comment:
will need to look into why the SDK generates so many pointer values for things that _SHOULD_ not be pointers